### PR TITLE
[ARCHETYPE-655] Get rid of Wagon API to download catalogs

### PR DESCRIPTION
**Pull request by slawekjaranowski**
https://issues.apache.org/jira/browse/ARCHETYPE-655

### CODE REVIEW

File: archetype-common/src/main/java/org/apache/maven/archetype/source/RemoteCatalogArchetypeDataSource.java [hunk 1]
```diff
@@ -77,303 +57,56 @@ public class RemoteCatalogArchetypeDataSource extends CatalogArchetypeDataSource
...
-
-        return wagon;
-    }
-
-    private AuthenticationInfo getAuthenticationInfo(String id) {
-        MavenSession session = legacySupport.getSession();
 
-        if (session != null && id != null) {
-            MavenExecutionRequest request = session.getRequest();
+        MetadataRequest request = new MetadataRequest();
```
Comment (cstamas) [inline archetype-common/src/main/java/org/apache/maven/archetype/source/RemoteCatalogArchetypeDataSource.java:64]: Neat trick, but I'd rather do it like here:
https://github.com/apache/maven/blob/master/maven-api-impl/src/main/java/org/apache/maven/internal/impl/DefaultTransportProvider.java
and
https://github.com/apache/maven/blob/master/maven-api-impl/src/main/java/org/apache/maven/internal/impl/DefaultTransportProvider.java

**Review by cstamas — COMMENTED** [review]
Looks good, but I'd not use metadata op to resolve catalog, as catalog is really not part of Resolver "layout" (and Resolver is oblivious about it). Here we deal with a resource "that happens" that is somewhere in remote repo (root), so I'd use same approach as in Maven4 Transporter implementation: direct use of Transport resolver interface.

**Comment by cstamas** [comment]
Don't get me wrong, I am _not -1 on this PR_, but anyone looking at the code will raise eyebrows when look into how is remote catalog fetched :smile:

**Review by cstamas — APPROVED** [review]
In fact, I am now contradicting myself: this PR is ok, and I myself was not aware of such use of Metadata, kudos. +1.

Leave it as is pls!

**Comment by HannesWell** [comment]
Is it intentional that this changes the logic how the `archetypeRepository` is selected among the specified `RemoteRepositoriey` in the `RemoteCatalogArchetypeDataSource`?

Previously a repo with `ARCHETYPE_REPOSITORY_ID` was always preferred:
https://github.com/apache/maven-archetype/blob/cc906677b5539b538c7ae640d894142ecf57fdc0/archetype-common/src/main/java/org/apache/maven/archetype/source/RemoteCatalogArchetypeDataSource.java#L102-L109

now always the first one one that is either `archetype` or `central` is selected:
https://github.com/apache/maven-archetype/blob/367a9fe79a6698c5a6fc1fbcbfb239819d5ba9c2/archetype-common/src/main/java/org/apache/maven/archetype/source/RemoteCatalogArchetypeDataSource.java#L90-L102

**Comment by jira-importer** [comment]
Resolve #655

**Pull request merged**